microprofile.jdt/com.redhat.microprofile.jdt.quarkus/src/main/java/com/redhat/microprofile/jdt/internal/quarkus/providers/QuarkusProjectIdentifier.java

```diff
@@ -0,0 +1,20 @@
+package com.redhat.microprofile.jdt.internal.quarkus.providers;
+
+import org.eclipse.jdt.core.IJavaProject;
+import org.eclipse.jdt.core.JavaModelException;
+
+import com.redhat.microprofile.jdt.core.IJavaProjectIdentifier;;
+
+public class QuarkusProjectIdentifier implements IJavaProjectIdentifier {
+
+	@Override
+	public boolean isProject(IJavaProject project) throws JavaModelException {
+		return project!= null && project.findType("io.quarkus.runtime.Application") != null;
```

Comment:
Move this code to the JDTQuarkusUtils and follow the same idea than
https://github.com/jbosstools/jbosstools-quarkus/blob/1c7ff8dd35fb64a842d9de8839fdc924eb36eb58/plugins/org.jboss.tools.quarkus.core/src/org/jboss/tools/quarkus/core/project/ProjectUtils.java#L183

It seems JBoss Tools check the existing of `io.quarkus.runtime.LaunchMode`

Reply:
Yes this was discussed with the Quarkus team and according to them Application is not part of the API and they recommended LaunchMode

Reply:
Thanks @jeffmaury for your feedback!